### ISSUE TITLE
Unexpected allocations in `point_jacobian!` in Julia 1.5.1:

### ISSUE THREAD

**Issue by rdeits** [JuliaRobotics/RigidBodyDynamics.jl#590]
From the unit tests on my machine with Julia 1.5.1:

```julia
point expressed in body frame: Test Failed at /home/rdeits/.julia/packages/RigidBodyDynamics/YeSNy/test/test_mechanism_algorithms.jl:378
  Expression: #= /home/rdeits/.julia/packages/RigidBodyDynamics/YeSNy/test/test_mechanism_algorithms.jl:378 =# @allocated(point_jacobian!(J_point, x, p, point)) == 0
   Evaluated: 32 == 0
```

**Comment by rdeits** [comment]
Accidentally closed in #591

**Comment by tkoolen** [comment]
Test fixed in https://github.com/JuliaRobotics/RigidBodyDynamics.jl/pull/605.
